doc/contributor/launchers.md

```diff
@@ -0,0 +1,38 @@
+# Executables in `bin` 
+
+## Requirements
+
+1.  [x] When TruffleRuby tar (GraalVM) is downloaded and unpacked, all files in `bin` and `language/ruby/bin` has to 
+    always run the TruffleRuby from the tar. 
+    -   Therefore `PATH` cannot be involved (`/usr/bin/env` cannot be used in shebang).
+    -   There cannot be absolute paths in `bin` executables.
+    -   All scripts in `bin` have to always resolve to `truffleruby` in the same dir.
+2.  [x] Newly installed gem executables has to always run the `truffleruby` from the same `bin`. 
+    -   By default RubyGems generate executables with absolute path.
+3.  [x] Behave as proper Ruby implementation when the `bin` is put on `PATH`.
+4.  [ ] RubyMine requires `gem` in `bin` to be Ruby (otherwise gems are not listed).
+    -   `gem` file in `bin` has to be Ruby or the `-x` option used by Rubymine has to be extended to understand our bash files in `bin`
```

Comment:
I think `-x` can naturally understand a bash/ruby file.
Say in `bin/gem`:
```bash
#!/usr/bin/env bash
exec "$(dirname $0)/truffleruby" -Sgem "$@"
#!/usr/bin/env ruby
exec "#{File.realpath(File.dirname(__FILE__))}/truffleruby", "-Sgem", *ARGV
```

Alternatively if we can assume the current Ruby invoked with -x is the right one then:
```bash
#!/usr/bin/env bash
exec "$(dirname $0)/truffleruby" -Sgem "$@"
#!/usr/bin/env ruby
load "#{File.expand_path('../../lib/bin/gem', File.realpath(__FILE__))}"
```
which avoids paying twice the startup cost.

It seems MRI has this behavior even without `-x`:
```bash
#!/usr/bin/env bash
echo bash
exit 0
#!/usr/bin/env ruby
puts "RUBY"
```

```bash
truffleruby/. (master) ± ./my_gem_x 
bash
truffleruby/. (master) ± ruby my_gem_x 
RUBY
```
@pitr-ch Can we use this to our advantage to avoid needing to generate `bin/` files?